Automattic/wp-calypso
# Signup: Domain step should show empty state when the query is emptied

#### Changes proposed in this Pull Request

Fresh start (no results) => clearing the search query while suggestions are loading renders empty state
Existing results => clearing renders previous returned domain suggestions

#### Testing instructions

1. Go to [/start/domains](https://container-ecstatic-hoover.calypso.live/start) and allow 3rd party cookies.
2. Start typing to search for a domain and while the results are still loading, clear the search query input
3. **UI should go back to empty state.**
4. If you cleared the input, type one character, **search shouldn't start and the empty state should prevail**.
5. Type in 2+ characters, search **should start and results should be shown as you'd expect**.

Demo: https://user-images.githubusercontent.com/14192054/124305375-af115900-db6d-11eb-99c6-b7d71bd966a9.mov

Fixes: https://github.com/Automattic/wp-calypso/issues/53752
